### PR TITLE
API-1835: Apply 02 more loops

### PR DESCRIPTION
**Pull request by deads2k**

### CODE REVIEW

File: pkg/operator/apiserver/controller/apiservice/apiservice_controller.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func NewAPIServiceController(
 			kubeInformersForNamespaces.InformersFor(metav1.NamespaceSystem).Core().V1().ConfigMaps().Informer(),
 			apiregistrationInformers.Apiregistration().V1().APIServices().Informer(),
 		)...,
-	).ToController(c.name, eventRecorder.WithComponentSuffix("apiservice-"+name+"-controller"))
+	).ToController(c.controllerInstanceName, eventRecorder.WithComponentSuffix("apiservice-"+instanceName+"-controller"))
```
Comment (p0lyn0mial) [inline pkg/operator/apiserver/controller/apiservice/apiservice_controller.go:80]: and events are not "persisted" so changing the name for this controller is fine.

File: pkg/operator/apiserver/controller/auditpolicy/auditpolicy_controller.go [hunk 2]
```diff
@@ -32,6 +35,7 @@ type auditPolicyController struct {
 // NewAuditPolicyController create a controller that watches the config.openshift.io/v1 APIServer object
 // and reconciles a ConfigMap in the target namespace with the audit.k8s.io/v1 policy.yaml file.
 func NewAuditPolicyController(
+	name string,
```
Comment (p0lyn0mial) [inline pkg/operator/apiserver/controller/auditpolicy/auditpolicy_controller.go:38]: nit: please rename to `instanceName`.

File: pkg/operator/apiserver/controller/nsfinalizer/finalizer_controller.go [hunk 3]
```diff
@@ -53,7 +53,10 @@ func NewFinalizerController(
 	return factory.New().ResyncEvery(time.Minute).WithSync(c.sync).WithInformers(
 		kubeInformersForTargetNamespace.Core().V1().Pods().Informer(),
 		kubeInformersForTargetNamespace.Apps().V1().DaemonSets().Informer(),
-	).ToController(fullname, eventRecorder.WithComponentSuffix("finalizer-controller"))
+	).ToController(
```
Comment (p0lyn0mial) [inline pkg/operator/apiserver/controller/nsfinalizer/finalizer_controller.go:56]: why? the controller doesn't use WithSyncDegradedOnError which is required for reporting a degraded condition on sync err, right ?

File: pkg/operator/apiserver/controller/workload/workload.go [hunk 4]
```diff
@@ -110,7 +110,10 @@ func NewController(name, operatorNamespace, targetNamespace, targetOperandVersio
 
 	return c.WithSync(controllerRef.sync).
 		WithInformers(informers...).
-		ToController(fmt.Sprintf("%sWorkloadController", name), eventRecorder)
+		ToController(
+			fmt.Sprintf("%sWorkloadController", name), // don't change what is passed here unless you also remove the old FooDegraded condition
```
Comment (p0lyn0mial) [inline pkg/operator/apiserver/controller/workload/workload.go:114]: why? the controller doesn't use WithSyncDegradedOnError which is required for reporting a degraded condition on sync err, right ?

File: pkg/operator/csi/csidrivernodeservicecontroller/csi_driver_node_service_controller.go [hunk 5]
```diff
@@ -115,13 +119,13 @@ func NewCSIDriverNodeServiceController(
 	).WithSyncDegradedOnError(
 		operatorClient,
 	).ToController(
-		c.name,
-		recorder.WithComponentSuffix("csi-driver-node-service_"+strings.ToLower(name)),
+		c.instanceName, // don't change what is passed here unless you also remove the old FooDegraded condition
```
Comment (p0lyn0mial) [inline pkg/operator/csi/csidrivernodeservicecontroller/csi_driver_node_service_controller.go:122]: this one actually uses `WithSyncDegradedOnError` so in this case we cannot change the name without a clean up routine.

**Comment by deads2k** [comment]
will get the comments in a followup.

**Pull request merged**